src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -607,8 +650,81 @@ private boolean isAllowedApiRequest(HttpServletRequest request) {
 			url=pathInfo;
 			isAllowedRequest=isApiFoundIntheList(url);
 		}
+
+		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_CROSSLIST)) && session.getAttribute(LAUNCH_TYPE).equals("lti")){
+			isAllowedRequest = isCrosslistAllowed(request, session, url);
+		}
+
 		return isAllowedRequest;
 	}
+
+	private boolean isCrosslistAllowed(HttpServletRequest request,
```

Comment:
Better to have separate methods for building / doing the call and processing the information returned.